### ISSUE TITLE
Eager load descendants?

### ISSUE THREAD

**Issue by dmaksimov** [lazychaser/laravel-nestedset#45]
Is there any way to eager load descendants without having to iterate through each node?

Right now I'm having to do 
```php
foreach ($nodes as $node) {
    $node->getDescendants();
}
```
It would be nice to do something like 
```php
$categories = Category::with('descendants')->get();
```

and the each category would have $category->descendants.

**Comment by lazychaser** [comment]
Do you still need a solution?

**Comment by rosswilson252** [comment]
A solution to this would be lovely.

**Comment by lazychaser** [comment]
@rosswilson252 what is the cost of loading the whole tree?

**Comment by dmaksimov** [comment]
There are instances when I wouldn't need to load the entire tree and loading the entire tree would use a lot more memory if there are a lot of records.

Something like the following would be great.

```
$items = MenuItems::with('descendants')->where('active', 1)->get();
```

Not sure how complicated it would be. Not really necessary since it'll probably not be used often but would be nice to have.

**Comment by lazychaser** [comment]
Now available in `4.1.0@dev`
